# HG changeset patch
# Parent 6bbb022a5a89d6a853ba7bd6eb99cd9ebe0f197a
# Date 1298820795 -3600

Use svn metadata in every case, use first tag from the svnrev

diff --git a/findversion.sh b/findversion.sh
--- a/findversion.sh
+++ b/findversion.sh
@@ -106,22 +106,18 @@
 	fi
 elif [ -d "$ROOT_DIR/.hg" ]; then
 	# We are a hg checkout
-	if [ -n "`hg status | grep -v '^?'`" ]; then
+	if [ -n "`hg status | grep -v '^?'`" ] || [ -z "`hg parent --template="{svnrev}"`" ]; then
 		MODIFIED="2"
 	fi
 	HASH=`LC_ALL=C hg id -i | cut -c1-12`
 	REV="h`echo $HASH | cut -c1-8`"
 	BRANCH="`hg branch | sed 's@^default$@@'`"
-	TAG="`hg id -t | grep -v 'tip$'`"
+	REV_NR=`hg log -f --template="{svnrev}\n" | grep -v "^$" | head -n 1`
+	TAG="`hg log -r "svnrev($REV_NR)" --template="{tags}" | grep -v 'tip$' | cut -f1 -d' '`"
 	if [ -n "$TAG" ]; then
 		BRANCH=""
 		REV="$TAG"
 	fi
-	REV_NR=`LC_ALL=C hg log -f -k "(svn r" -l 1 --template "{desc|firstline}\n" | grep "^(svn r[0-9]*)" | sed "s@.*(svn r\([0-9]*\)).*@\1@"`
-	if [ -z "$REV_NR" ]; then
-		# No rev? Maybe it is a custom hgsubversion clone
-		REV_NR=`LC_ALL=C hg parent --template="{svnrev}"`
-	fi
 elif [ -f "$ROOT_DIR/.ottdrev" ]; then
 	# We are an exported source bundle
 	cat $ROOT_DIR/.ottdrev
